pkg/kubelet/dockershim/libdocker/kube_docker_client.go

```diff
@@ -489,7 +490,28 @@ func (d *kubeDockerClient) StartExec(startExec string, opts dockertypes.ExecStar
 		sopts.ExecStarted <- struct{}{}
 	}
 
-	return d.holdHijackedConnection(sopts.RawTerminal || opts.Tty, sopts.InputStream, sopts.OutputStream, sopts.ErrorStream, resp)
+	err = d.holdHijackedConnection(sopts.RawTerminal || opts.Tty, sopts.InputStream, sopts.OutputStream, sopts.ErrorStream, resp, timeout)
+	// Kill the process if timeout elapsed
+	if _, ok := err.(operationTimeout); ok {
+		inspect, err2 := d.InspectExec(startExec)
+		if err2 != nil {
+			glog.Errorf("Unable to inspect exec session %s in container %s for timeout termination!", startExec, inspect.ContainerID)
```

Comment:
suggest something like `glog.Errorf("failed to kill exec process after timeout: session: %s, container: %s, error: %v", startExec, inspect.ContainerID, inspectError)`

Regarding `inspect.ContainerID`, can you trust that `inspect` holds a value when you get an error here?

Reply:
similar suggestions below